src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -3,33 +3,147 @@
  * @author Box
  */
 import * as React from 'react';
-import flow from 'lodash/flow';
 import { FormattedMessage } from 'react-intl';
+import flow from 'lodash/flow';
+import getProp from 'lodash/get';
+import { MetadataEmptyState } from '@box/metadata-editor';
+import type { MessageDescriptor } from 'react-intl';
+import messages from '../common/messages';
+import { normalizeTemplates } from '../../features/metadata-instance-editor/metadataUtil';
+import { isUserCorrectableError } from '../../utils/error';
 import { withAPIContext } from '../common/api-context';
 import { withErrorBoundary } from '../common/error-boundary';
 import { withLogger } from '../common/logger';
-import { ORIGIN_METADATA_SIDEBAR_REDESIGN } from '../../constants';
-import { EVENT_JS_READY } from '../common/logger/constants';
 import { mark } from '../../utils/performance';
-import messages from '../common/messages';
+import { EVENT_JS_READY } from '../common/logger/constants';
+import API from '../../api';
+import {
+    FIELD_IS_EXTERNALLY_OWNED,
+    FIELD_PERMISSIONS,
+    FIELD_PERMISSIONS_CAN_UPLOAD,
+    IS_ERROR_DISPLAYED,
+    ORIGIN_METADATA_SIDEBAR_REDESIGN,
+} from '../../constants';
+import type { ElementsXhrError, ErrorContextProps } from '../../common/types/api';
+import type { BoxItem } from '../../common/types/core';
+import type { MetadataEditor, MetadataTemplate } from '../../common/types/metadata';
+import type { WithLoggerProps } from '../../common/types/logging';
 
 import './MetadataSidebarRedesign.scss';
 
+interface ExternalProps {
+    isBoxAiSuggestionsEnabled: boolean;
+    isFeatureEnabled: boolean;
+    selectedTemplateKey?: string;
+    templateFilters?: Array<string> | string;
+}
+
+interface PropsWithoutContext extends ExternalProps {
+    elementId: string;
+    fileId: string;
+    hasSidebarInitialized?: boolean;
+}
+
+// For some reason, when changing this type to an interface, the MetadataSidebarRedesign component throws an error on the onError prop.
+export type MetadataSidebarRedesignProps = {
+    api: API;
+} & PropsWithoutContext &
+    ErrorContextProps &
+    WithLoggerProps;
+
+interface Metadata {
+    editors: Array<MetadataEditor>;
+    templates: Array<MetadataTemplate>;
+}
+
 const MARK_NAME_JS_READY = `${ORIGIN_METADATA_SIDEBAR_REDESIGN}_${EVENT_JS_READY}`;
 
 mark(MARK_NAME_JS_READY);
 
-function MetadataSidebarRedesign() {
+const MetadataSidebarRedesign: React.FunctionComponent<MetadataSidebarRedesignProps> = ({
```

Comment:
There's a lot of of business logic defined in this component. Should we extract some of it out to a hook and let this component focus mostly on it's presentation?

Reply:
created the hook and moved that to /hooks folder just as in the other PR https://github.com/box/box-ui-elements/pull/3606